src/main/java/de/hysky/skyblocker/events/LocationEvents.java

```diff
@@ -0,0 +1,22 @@
+package de.hysky.skyblocker.events;
+
+import de.hysky.skyblocker.utils.Location;
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+@Environment(EnvType.CLIENT)
+public class LocationEvents {
+    public static final Event<LocationChange> CHANGE =
+            EventFactory.createArrayBacked(LocationChange.class, callbacks -> location -> {
+                for (LocationChange callback : callbacks)
+                    callback.onLocationChange(location);
+            });
+
+    @Environment(EnvType.CLIENT)
+    @FunctionalInterface
+    public interface LocationChange {
+        void onLocationChange(Location location);
```

Comment:
Rename the `location` parameter to better describe what information it provides as it's a big ambiguous as to whether its the new location or the old one.